src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -141,14 +141,31 @@ internal IEnumerable<FileInfo> GetRestoreFiles()
 
         internal IEnumerable<FileInfo> GetBuildFiles()
         {
-            return GetInventory(_inventoryFiles.Build, GetRestoreFiles, DoBuild);
+            return GetInventory(
+                _inventoryFiles.Build,
+                () =>
+                {
+                    var preInventory = new List<FileInfo>(GetRestoreFiles());
+                    preInventory.AddRange(GetSourceFiles());
+                    return preInventory;
+                },
+                DoBuild);
         }
 
-        private IEnumerable<FileInfo> GetInventory(FileInfo file, Func<IEnumerable<FileInfo>> beforeAction, Action action)
+        private IEnumerable<FileInfo> GetInventory(
+            FileInfo file,
+            Func<IEnumerable<FileInfo>> beforeAction,
+            Action action)
         {
+            var inventory = Enumerable.Empty<FileInfo>();
             if (file.Exists)
             {
-                return LoadInventory(file);
+                inventory = LoadInventory(file);
+            }
+
+            if(inventory.Any())
```

Comment:
How so? That's exactly the problem, the restore file is being empty when it shouldn't.